src/core/lv_obj_tree.c

```diff
@@ -235,6 +235,27 @@ void lv_obj_move_background(lv_obj_t * obj)
     lv_obj_invalidate(parent);
 }
 
+void lv_obj_swap(lv_obj_t * obj1, lv_obj_t * obj2)
+{
+    LV_ASSERT_OBJ(obj1, MY_CLASS)
+    LV_ASSERT_OBJ(obj2, MY_CLASS)
+
+    lv_obj_t* parent = lv_obj_get_parent(obj1);
+    lv_obj_t* parent2 = lv_obj_get_parent(obj2);
+    LV_ASSERT(parent == parent2);
+
+    uint_fast32_t index1 = lv_obj_get_child_id(obj1);
+    uint_fast32_t index2 = lv_obj_get_child_id(obj2);
+
+    parent->spec_attr->children[index1] = obj2;
+    parent->spec_attr->children[index2] = obj1;
+
+    lv_event_send(parent, LV_EVENT_CHILD_CHANGED, obj2);
```

Comment:
`LV_EVENT_CHILD_CHANGED` should be sent to the parent.

It'd be better to send 2 events only if the parents are different.